From bcd325b434cf5d0328c342f9473c1f1d7d6146d5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 21 Nov 2017 08:05:38 +0000
Subject: [PATCH] Support an alternative bash build than the default

This change will have dracut use a specialised build of bash from the
new `bash-recovery` package in Solus within the initramfs as opposed to
the full fat glibc-based bash binary.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 modules.d/00bash/module-setup.sh | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/modules.d/00bash/module-setup.sh b/modules.d/00bash/module-setup.sh
index 89ef6541..285b1972 100755
--- a/modules.d/00bash/module-setup.sh
+++ b/modules.d/00bash/module-setup.sh
@@ -24,9 +24,17 @@ depends() {
 # Install the required file(s) and directories for the module in the initramfs.
 install() {
 
-    inst /bin/bash
-
-    # Prefer bash as default shell if no other shell is preferred.
-    [[ -L $initdir/bin/sh ]] || ln -sf bash "${initdir}/bin/sh"
+    inst /usr/bin/bash
+
+    # Prefer bash.recovery binary due to smaller size
+    if [[ -x /usr/bin/bash.recovery ]]; then
+        inst /usr/bin/bash.recovery
+        ln -sf /usr/bin/bash.recovery "${initdir}/bin/sh"
+        ln -sf /usr/bin/bash.recovery "${initdir}/bin/bash"
+        ln -sf /usr/bin/bash.recovery "${initdir}/usr/bin/bash"
+    else
+        # Prefer bash as default shell if no other shell is preferred.
+        [[ -L $initdir/bin/sh ]] || ln -sf bash "${initdir}/bin/sh"
+    fi
 
 }
-- 
2.43.0

